TownSuite.Web.ImageGen/HeifDecoder.cs

```diff
@@ -9,10 +9,14 @@ public static class HeifDecoder
     {
         public static Image<Rgba32> ConvertHeifToSharp(Stream stream)
         {
-            using var ms = new MemoryStream();
-            stream.CopyTo(ms);
-   
-            using var heifContext = new HeifContext(ms.ToArray());
+            if (!Available())
+                throw new Exception("HeifDecoder is not available");
+            if (stream == null || !stream.CanRead)
+                throw new ArgumentException("Invalid stream", nameof(stream));
```

Comment:
nameof here will only get the name of the variable, not the type (in this case it will always be "stream"). If that is not the intention you could use 
```C#
stream.GetType().Name
```
You would need to handle the case where the stream is null in some way if you go with that option.

Reply:
Just added a commit to address this